src/newgrf_roadstop.h

```diff
@@ -0,0 +1,189 @@
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file newgrf_roadstop.h NewGRF definitions and structures for road stops.
+ */
+
+#ifndef NEWGRF_ROADSTATION_H
+#define NEWGRF_ROADSTATION_H
+
+#include "newgrf_animation_type.h"
+#include "newgrf_spritegroup.h"
+#include "newgrf_class.h"
+#include "newgrf_commons.h"
+#include "newgrf_town.h"
+#include "road.h"
+
+/** The maximum amount of roadstops a single GRF is allowed to add */
+static const int NUM_ROADSTOPS_PER_GRF = 255;
+
+enum RoadStopClassID : byte {
+	ROADSTOP_CLASS_BEGIN = 0,    ///< The lowest valid value
+	ROADSTOP_CLASS_DFLT = 0,     ///< Default road stop class.
+	ROADSTOP_CLASS_WAYP,         ///< Waypoint class.
+	ROADSTOP_CLASS_MAX = 255,    ///< Maximum number of classes.
+};
+DECLARE_POSTFIX_INCREMENT(RoadStopClassID)
+
+/* Some Triggers etc. */
+enum RoadStopRandomTrigger {
+	RSRT_NEW_CARGO,       ///< Trigger roadstop on arrival of new cargo.
+	RSRT_CARGO_TAKEN,     ///< Trigger roadstop when cargo is completely taken.
+	RSRT_VEH_ARRIVES,     ///< Trigger roadstop when road vehicle arrives.
+	RSRT_VEH_DEPARTS,     ///< Trigger roadstop when road vehicle leaves.
+	RSRT_VEH_LOADS,       ///< Trigger roadstop when road vehicle loads.
+};
+
+/**
+ * Various different options for availability, restricting
+ * the roadstop to be only for busses or for trucks.
+ */
+enum RoadStopAvailabilityType : byte {
+	ROADSTOPTYPE_PASSENGER,    ///< This RoadStop is for passenger (bus) stops.
+	ROADSTOPTYPE_FREIGHT,      ///< This RoadStop is for freight (truck) stops.
+	ROADSTOPTYPE_ALL,          ///< This RoadStop is for both types of station road stops.
+
+	ROADSTOPTYPE_END,
+};
+
+/**
+ * Different draw modes to disallow rendering of some parts of the stop
+ * or road.
+ */
+enum RoadStopDrawMode : byte {
+	ROADSTOP_DRAW_MODE_NONE        = 0,
+	ROADSTOP_DRAW_MODE_ROAD        = 1 << 0, ///< Bay stops: Draw the road itself
+	ROADSTOP_DRAW_MODE_OVERLAY     = 1 << 1, ///< Drive-through stops: Draw the road overlay, e.g. pavement
+};
+DECLARE_ENUM_AS_BIT_SET(RoadStopDrawMode)
+
+enum RoadStopSpecFlags {
+	RSF_CB141_RANDOM_BITS       = 0, ///< Callback 141 needs random bits.
+	RSF_NO_CATENARY             = 2, ///< Do not show catenary.
+	RSF_DRIVE_THROUGH_ONLY      = 3, ///< Stop is drive-through only.
+	RSF_NO_AUTO_ROAD_CONNECTION = 4, ///< No auto road connection.
+	RSF_BUILD_MENU_ROAD_ONLY    = 5, ///< Only show in the road build menu (not tram).
+	RSF_BUILD_MENU_TRAM_ONLY    = 6, ///< Only show in the tram build menu (not road).
+};
+
+/** Scope resolver for road stops. */
+struct RoadStopScopeResolver : public ScopeResolver {
+	TileIndex tile;                             ///< %Tile of the station.
```

Comment:
Typo?
```suggestion
	TileIndex tile;                             ///< Tile of the station.
```

Reply:
See the various other existing ScopeResolver/ResolverObject subclasses